firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -4791,8 +4796,8 @@ extension BrowserViewController: TopTabsDelegate {
     }
 
     func topTabsDidPressNewTab(_ isPrivate: Bool) {
-        let shouldLoadCustomHomePage = isToolbarRefactorEnabled && NewTabAccessors.getHomePage(profile.prefs) == .homePage
-        let homePageURL = HomeButtonHomePageAccessors.getHomePage(profile.prefs)
+        let shouldLoadCustomHomePage = isToolbarRefactorEnabled && newTabSettings == .homePage
+        let homePageURL = NewTabHomePageAccessors.getHomePage(profile.prefs)
```

Comment:
@FilippoZazzeroni This is the change that is causing the tests to fail. I'm not too familiar with this area, but I thought `topTabs` was related to the iPad version and not iPhone, so curious to see whh the modifications here are impacting the iPhone version. 

The reason that this is breaking is because the tests assumes that tapping on the `new tab` button should open what's been configured under the homepage settings, which is similar to the previous old `home` button.

I'm not sure what the requirement is for this work, but this PR introduced the change in that broke the tests. So we may want to revert changes here or update tests to match new requirements.

cc: @dragosb01

Reply:
@FilippoZazzeroni any issues if I replace `newTabSettings` here with `NewTabAccessors.getHomePage(profile.prefs)`?